### DISCUSSION TITLE
`useReducer` state as RHF values

### DISCUSSION THREAD

**Discussion by Dragomir-Ivanov** [react-hook-form/react-hook-form#8148] [Q&A] [answered]
Hi,
I have a form, with several inputs, that contain numbers. When change certain field, it might modify some of other fields. In order to contain the calculation model in one place, I choose to make it a `reducer` function. Potentially this makes `reset` the form, on every keystroke of these fields.

I have an example, but whatever I do, the whole form gets re-render ( with all its fields).
https://codesandbox.io/s/zen-lichterman-ueeij0?file=/src/App.js

Doesn't `reset` compares with previous values, and renders its `input` elements, only if the new value is different from the old one?

And overall, is this a good strategy with RHF, or perhaps there is another more efficient way to do it?

Thank you

**Comment by Moshyfawn** [comment, marked as answer]
`reset` form method resets the whole form by design

[From RHF docs](https://react-hook-form.com/api/useform/reset):
> Reset the entire form state, fields reference, and subscriptions. 

If you want to reset a specific field, you can use
- either [the `resetField` form method](https://react-hook-form.com/api/useform/resetfield)
- or [the `setValue` form method](https://react-hook-form.com/api/useform/setvalue)

RHF doesn't manage your UI updates. It's a headless hook that manipulates the form's logic

**Reply by Dragomir-Ivanov** [reply]
Thank you, for your fast reply!

So basically, I will need to make a diff between the two `values` and update only those paths with `setValue`.
Is there any way I can make this with RHF, in a more idiomatic way?
